### PR TITLE
Fix MSVC compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
* reg. not all control paths return a value (C4715)
* reg. unreferenced local variable (C4101)
* reg. discarding return value of function with 'nodiscard' attribute (C4834) (apparently a bug)
* reg. check operator precedence for possible error; use parentheses to clarify precedence (C4554)  (actually a bug, `&` vs. `&&`!)
* reg. was declared deprecated (C4996) (in thirdparty, hence disabling the warning)
* reg. possible loss of data (C4244)  (in thirdparty, hence disabling the warning, unfortunately it doesn't help!?!)
* reg. differs in indirection to slightly different base types (C4057) (in thirdparty, hence disabling the warning)
* reg. result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?) (C4334)  (in thirdparty, hence disabling the warning)

### CODE REVIEW

**Comment by igorkorsukov** [comment]
@lpugin FYI

**Review by igorkorsukov — APPROVED** [review]

File: src/importexport/mei/internal/meiconverter.cpp [hunk 1]
```diff
@@ -1348,7 +1348,7 @@ Convert::PitchStruct Convert::pitchFromMEI(const libmei::Note& meiNote, const li
 
     int step = meiNote.HasPname() ? meiNote.GetPname() - 1 : 0;
     // It should never be necessay, but just in case
-    std::clamp(step, 0, 6);
+    (void)std::clamp(step, 0, 6);
```
Comment (cbjeukendrup) [inline src/importexport/mei/internal/meiconverter.cpp:1351]: I think this is rather intended to be:
```suggestion
    step = std::clamp(step, 0, 6);
```

Comment (Jojo-Schmitz) [reply src/importexport/mei/internal/meiconverter.cpp]: thanks, fixed

File: thirdparty/flac/CMakeLists.txt [hunk 2]
```diff
@@ -71,6 +71,8 @@ set(MODULE_USE_PCH_NONE ON)
 set(MODULE_USE_UNITY_NONE ON)
 include(${PROJECT_SOURCE_DIR}/build/module.cmake)
 
+target_no_warning(${MODULE} -Wno-deprecated)
+target_no_warning(${MODULE} -Wno-conversion)
```
Comment (Jojo-Schmitz) [inline thirdparty/flac/CMakeLists.txt:75]: Strangely this doesnt help, the 2 warnings still show:
Warning	C4244	'=': conversion from 'const uint32_t' to 'FLAC__byte', possible loss of data	...\thirdparty\flac\flac-1.4.3\src\libFLAC\metadata_iterators.c	2725
Warning	C4244	'=': conversion from 'uint32_t' to 'FLAC__byte', possible loss of data	...\thirdparty\flac\flac-1.4.3\src\libFLAC\metadata_iterators.c	2877

File: src/importexport/mei/internal/meiexporter.cpp [hunk 3]
```diff
@@ -132,6 +132,7 @@ bool MeiExporter::write(QIODevice& destinationDevice)
         out << String::fromStdString(strStream.str());
     }
     catch (char* str) {
+        (void)str;
```
Comment (lpugin) [inline src/importexport/mei/internal/meiexporter.cpp:135]:
```suggestion
        UNUSED(str);
```

alternatively we could probably remove the variable name
```cpp
catch (char*) {
    return false;
}
```
I don't know if some compilers would complain about that

Comment (Jojo-Schmitz) [reply src/importexport/mei/internal/meiexporter.cpp]: Yes, UNUSED is better, I thought I tried it and saw it not working. May need new glasses...

Comment (Jojo-Schmitz) [reply src/importexport/mei/internal/meiexporter.cpp]: As per https://en.cppreference.com/w/cpp/language/try_catch `catch (char*)` is not permitted, unless I missread it.
Anyway, why taking chances ;-)

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**